### PR TITLE
NLB-ALB support

### PR DESCRIPTION
**Pull request by haggaret**
- Removed reliance on specifying Beanstalk app name (and corresponding beanstalk API calls)
- Added support for an NLB/ALB pair (to support static IPs for load balancers)
- updated to python 3
- Added new optional parameter to display additional information beside each app name

### CODE REVIEW

File: app/awslib.py [hunk 1]
```diff
@@ -4,43 +4,36 @@
 import os
 import socket
 
-# List of all EIPs
-def _list_eips(region, filter):
+
+# List of all EIPs in the given region
+def list_eips(region, filter):
     print ("Connecting to ec2...")
     client = boto3.client('ec2', region_name=region)
```
Comment (miketypeguy) [inline app/awslib.py:12]: Perhaps we should handle if this fails?

File: app/awslib.py [hunk 2]
```diff
@@ -49,74 +42,186 @@ def _active_balancer(dns_name, region):
...
+# Get a file from S3
+def get_file(bucket_name, s3_path, local_path):
     if os.path.isfile(local_path):
         print ("Deleting current file...")
         os.remove(local_path)
         print ("Done")
     print ("Retrieving config file...")
     s3 = boto3.resource('s3')
-    s3.meta.client.download_file(bucket_name, s3_path, local_path)
+    s3.Bucket(bucket_name).download_file(s3_path, local_path)
```
Comment (miketypeguy) [inline app/awslib.py:233]: How do we know that this succeeded?

File: app/awslib.py [hunk 3]
```diff
@@ -49,74 +42,186 @@ def _active_balancer(dns_name, region):
...
+    # s3.meta.client.download_file(bucket_name, s3_path, local_path)
```
Comment (miketypeguy) [inline app/awslib.py]: Do we need this?

File: app/awslib.py [hunk 4]
```diff
@@ -49,74 +42,186 @@ def _active_balancer(dns_name, region):
...
-                        if 'PublicIpAddress' in instance:
-                            instance_ips.append(instance['PublicIpAddress'])
-                        else:
-                            print ("The instance %s has no public IP" % str(instance['InstanceId']))
+                instance_ips.extend(_get_instances_public_ip(ec2_client, instances))
+    # Only look at v2 load balancers if we haven't already found the load balancer above
+    if not lb_found:
+        print("Didn't find the load balancer in the list of classic load balancers")
+        print ("Connecting to ec2 elb v2...")
+        elbv2 = boto3.client('elbv2', region_name=region)
```
Comment (miketypeguy) [inline app/awslib.py]: maybe check this as well as it could possible mess up the later calls

File: app/awslib.py [hunk 5]
```diff
@@ -49,74 +42,186 @@ def _active_balancer(dns_name, region):
...
-    print ("Connected!")
-    print ("Retrieving load balancers...")
-    all_lbs = elb.describe_load_balancers()['LoadBalancerDescriptions']
+def list_instance_ips(lb_name, region):
+    print ("Getting IPs for instances behind load balancer %s..." % lb_name)
     instance_ips = []
-    for lb in all_lbs:
+    lb_found = False
+    print ("Connecting to ec2 elb v1...")
+    elbv1 = boto3.client('elb', region_name=region)
```
Comment (miketypeguy) [inline app/awslib.py:175]: Check as well?

**Review by miketypeguy — CHANGES_REQUESTED** [review]
Possibly check some results of boto calls to make sure they didn't error out

File: app/views.py [hunk 6]
```diff
@@ -210,6 +234,11 @@ def handle_app(appname):
         return jsonify(**eval(line))
 
 
+@app.route('/<appname>/all')
```
Comment (haggaret) [inline app/views.py]: After chatting with @ryan-j-mci about this, I think I'm going to remove the /all functionality - don't think we need it.

File: app/awslib.py (hunk 1 above)
Comment (haggaret) [reply app/awslib.py:12]: Yeah sure, easy enough.

File: app/awslib.py (hunk 2 above)
Comment (haggaret) [reply app/awslib.py:233]: if the file exists? :)

File: app/awslib.py (hunk 3 above)
Comment (haggaret) [reply app/awslib.py]: The commented out code? Nope.

File: app/awslib.py (hunk 4 above)
Comment (haggaret) [reply app/awslib.py]: Sure.

File: app/awslib.py (hunk 5 above)
Comment (haggaret) [reply app/awslib.py:175]: Sure

File: app/views.py (hunk 6 above)
Comment (miketypeguy) [reply app/views.py]: Cool

File: app/awslib.py (hunk 2 above)
Comment (miketypeguy) [reply app/awslib.py:233]: Yeah looks like the some of the instances where this method is called do actually check when they try to open the file.  Some do not.

**Review by miketypeguy — APPROVED** [review]

**Pull request merged**